packages/web3-core/src/types.ts

```diff
@@ -40,6 +43,16 @@ export type Web3BaseProviderConstructor = new <API extends Web3APISpec>(
 	net?: Socket,
 ) => Web3BaseProvider<API>;
 
+// TODO: When we have `web3-types` package we can share TransactionType
+export type TransactionTypeParser = (transaction: Record<string, unknown>) => HexString | undefined;
+
+// TODO: When we have `web3-types` package we can share TransactionType
+export type TransactionBuilder = (options: {
```

Comment:
I'm working on breaking all the methods in `eth_tx.ts` into separate files, and was wanting to create a `transactionBuilder` method that uses `defaultTransactionBuilder` if not overridden like you have setup for `transactionTypeParser`. In doing so, I'm wondering why you chose to pass a object to `TransactionBuilder` instead of just passing the parameters themselves?

Reply:
For more  context, I'd like to create `transactionBuilder` like so:

```typescript
export const transactionBuilder = async (
    transaction: Transaction,
	web3Context: Web3Context<EthExecutionAPI>,
	privateKey?: HexString | Buffer,
    transactionBuilder: TransactionBuilder = defaultTransactionBuilder,
) => transactionBuilder({
    transaction: transaction as unknown as Record<string, unknown>,
    web3Context,
    privateKey
});
```

Reply:
I'm realizing you already have similar functionality [here](https://github.com/ChainSafe/web3.js/blob/f932da4f8544ba3f9f07c6981abdb3b1def2d993/packages/web3-eth/src/eth_tx.ts#L266), but I think the `defaultTransactionBuilder` is an implementation detail that a dev using `transactionBuilder` shouldn't have to worry about